BaragonService/src/main/java/com/hubspot/baragon/service/edgecache/cloudflare/CloudflareEdgeCache.java

```diff
@@ -0,0 +1,96 @@
+package com.hubspot.baragon.service.edgecache.cloudflare;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.hubspot.baragon.models.BaragonRequest;
+import com.hubspot.baragon.service.edgecache.EdgeCache;
+import com.hubspot.baragon.service.edgecache.cloudflare.client.CloudflareClient;
+import com.hubspot.baragon.service.edgecache.cloudflare.client.CloudflareClientException;
+import com.hubspot.baragon.service.edgecache.cloudflare.client.CloudflareDnsRecord;
+import com.hubspot.baragon.service.edgecache.cloudflare.client.CloudflareZone;
+
+import io.dropwizard.setup.Environment;
+
+public class CloudflareEdgeCache implements EdgeCache {
+  private static final Logger LOG = LoggerFactory.getLogger(CloudflareEdgeCache.class);
+
+  private static final String CACHE_TAG_FORMAT = "staticjsapp-%s-%s";
+
+  private final CloudflareClient cf;
+  private final Environment environment;
+
+  @Inject
+  public CloudflareEdgeCache(CloudflareClient cf, Environment environment) {
+    this.cf = cf;
+    this.environment = environment;
+  }
+
+  /**
+   * Invalidation will eventually occur when the TTL expires, so it's not a showstopper if this fails.
+   */
+  @Override
+  public boolean invalidateIfNecessary(BaragonRequest request) {
+    if (request.getLoadBalancerService().getOptions().get("edgeCacheDNS") == null) {
+      return false;
+    }
+
+    try {
+      String edgeCacheDNS = ((String) request.getLoadBalancerService().getOptions().get("edgeCacheDNS"));
```

Comment:
would it make more sense to make this an explicit field on the BaragonService instead of having it be something in the options? The options have generally been used with the purpose of injecting data into the templates to be rendered, which is why it is a <String, Object> instead of something more specific. If we have an optional field on the baragon service, it's then a much cleaner call like `service.getEdgeCacheDNS.isPresent()`.

Also, I could see a case where an app is hosted on multiple domains (multiple Baragon groups) and might need multiple things cleared in which case it'd need to be more of a list of domains. Do we expect to need to handle that use case?